package/MDAnalysis/lib/distances.py

```diff
@@ -1432,25 +1493,62 @@ def calc_bonds(coords1: Union[np.ndarray, 'AtomGroup'],
     .. versionchanged:: 2.3.0
        Can now accept an :class:`~MDAnalysis.core.groups.AtomGroup` as an
        argument in any position and checks inputs using type hinting.
+    .. versionchanged:: 2.4.0
+       Can now optionally use the fast distance functions from distopia
     """
     numatom = coords1.shape[0]
     bondlengths = _check_result_array(result, (numatom,))
 
     if numatom > 0:
         if box is not None:
             boxtype, box = check_box(box)
-            if boxtype == 'ortho':
-                _run("calc_bond_distance_ortho",
-                     args=(coords1, coords2, box, bondlengths),
-                     backend=backend)
+            if boxtype == "ortho":
+                if backend == "distopia":
+                    # need explicit branch on backend to prepare input types correctly
+                    # must assign to result to get correct reference on memview
+                    bondlengths = _run(
+                        "calc_bonds_ortho_float",
+                        args=(coords1, coords2, box[:3]),
+                        kwargs={"results": bondlengths.astype(np.float32)},
+                        backend=backend,
+                    )
+                    # upcast is currently required, change for 3.0
+                    bondlengths = bondlengths.astype(np.float64)
```

Comment:
rather than have extra branches here, would it make more sense to have a stub `._distopia.py` which does the logic in this branch.

this stub could also redirect calls that aren't currently supported in distopia (anything but calc_bonds) back to the "serial" backend

Reply:
Yeah I'm +1 this - your mypy issues are mostly down to trying to shoehorn in a Callable into an ndarray. It would be a lot better if the behaviour of the `_run` calls was the same across all the backends, rather than have some that return things and some that do inplace changes to bondlengths.

Reply:
Done now with a stub